src/Core/Scheduler/Providers/WMS_Provider.js

```diff
@@ -74,26 +73,41 @@ WMS_Provider.prototype.tileInsideLimit = function tileInsideLimit(tile, layer) {
     return tile.level >= layer.options.zoom.min && tile.level <= layer.options.zoom.max && layer.extent.intersect(tile.extent);
 };
 
-WMS_Provider.prototype.getColorTexture = function getColorTexture(tile, layer) {
+WMS_Provider.prototype.getColorTexture = function getColorTexture(tile, layer, targetLevel) {
     if (!this.tileInsideLimit(tile, layer)) {
         return Promise.reject(`Tile '${tile}' is outside layer bbox ${layer.extent}`);
     }
     if (tile.material === null) {
         return Promise.resolve();
     }
 
-    const coords = tile.extent.as(layer.projection);
+    let extent = tile.extent;
+    if (targetLevel != undefined) {
```

Comment:
nits: can we invert this if-else ? (I find positive logic more readable)

Reply:
Done